src/EnergyPlus/api/datatransfer.cc

```diff
@@ -229,34 +247,71 @@ int getActuatorHandle(const char* componentType, const char* controlType, const
 
 void resetActuator(int handle) {
     // resets the actuator so that E+ will use the internally calculated value again
-    auto & theActuator(EnergyPlus::DataRuntimeLanguage::EMSActuatorAvailable(handle));
-    *theActuator.Actuated = false;
+    if (handle >= 1 && handle <= (int)EnergyPlus::DataRuntimeLanguage::EMSActuatorAvailable.size()) {
+        auto & theActuator(EnergyPlus::DataRuntimeLanguage::EMSActuatorAvailable(handle));
+        *theActuator.Actuated = false;
+    } else {
+        if (EnergyPlus::DataGlobals::eplusRunningViaAPI) {
+            throw std::runtime_error("Actuator handle out of range in resetActuator");
+        } else {
+            // must be running from python plugin, need to fatal out once the plugin is done
+            // throw an error, set the fatal flag, and then return
+            EnergyPlus::ShowSevereError("Data Exchange API: index error in resetActuator; received handle: " + std::to_string(handle));
+            EnergyPlus::ShowContinueError("The resetActuator function will return to allow the plugin to finish, then EnergyPlus will abort");
+            EnergyPlus::PluginManagement::shouldIssueFatalAfterPluginCompletes = true;
+        }
+    }
 }
 
 void setActuatorValue(const int handle, const Real64 value) {
-    auto & theActuator(EnergyPlus::DataRuntimeLanguage::EMSActuatorAvailable(handle));
-    if (theActuator.RealValue) {
-        *theActuator.RealValue = value;
-    } else if (theActuator.IntValue) {
-        *theActuator.IntValue = (int)std::lround(value);
+    if (handle >= 1 && handle <= (int)EnergyPlus::DataRuntimeLanguage::EMSActuatorAvailable.size()) {
+        auto & theActuator(EnergyPlus::DataRuntimeLanguage::EMSActuatorAvailable(handle));
+        if (theActuator.RealValue) {
+            *theActuator.RealValue = value;
+        } else if (theActuator.IntValue) {
+            *theActuator.IntValue = (int)std::lround(value);
+        } else {
+            // follow protocol from EMS manager, where 1.0 is true, 0.0 is false, and anything else is also false
+            *theActuator.LogValue = value < 0.99999 && value < 1.00001; // allow small tolerance while passing between languages and types
```

Comment:
LogValue is bool. If value < 0.99999 it should be false, here it's set to true. I think you meant value > 0.99999 ?

Reply:
Outstanding find @rraustad, thanks!

Reply:
No need to do anything now but after thinking about this I think value > 0.99999 is sufficient. If value is ever > 1.00001 then that's a mistake and no need to waste the clock.